index.js

```diff
@@ -51,50 +53,66 @@ module.exports = (robot, options) => {
     // Wait a random delay to more evenly distribute requests
     const delay = options.delay ? options.interval * Math.random() : 0
 
-    robot.log.debug({repository, delay, interval: options.interval}, `Scheduling interval`)
+    app.log.info(
+      { repository, delay, interval: options.interval },
+      `Scheduling interval`
+    )
 
     intervals[repository.id] = setTimeout(() => {
       const event = {
         name: 'schedule',
-        payload: {action: 'repository', installation, repository}
+        payload: { action: 'repository', installation, repository }
       }
 
       // Trigger events on this repository on an interval
-      intervals[repository.id] = setInterval(() => robot.receive(event), options.interval)
+      intervals[repository.id] = setInterval(
+        () => app.receive(event),
+        options.interval
+      )
 
       // Trigger the first event now
-      robot.receive(event)
+      app.receive(event)
     }, delay)
   }
 
   async function eachInstallation (callback) {
-    robot.log.trace('Fetching installations')
-    const github = await robot.auth()
+    app.log.trace('Fetching installations')
+    const github = await app.auth()
 
-    const req = github.apps.getInstallations({per_page: 100})
-    await github.paginate(req, res => {
-      (options.filter ? res.data.filter(inst => options.filter(inst)) : res.data)
-        .forEach(callback);
-    })
+    const installations = await github.paginate(
+      github.apps.listInstallations.endpoint.merge({ per_page: 100 })
+    )
+
+    const filteredInstallations = options.filter
+      ? installations.filter(inst => options.filter(inst))
+      : installations
+    return filteredInstallations.forEach(callback)
   }
 
   async function eachRepository (installation, callback) {
-    robot.log.trace({installation}, 'Fetching repositories for installation')
-    const github = await robot.auth(installation.id)
-
-    const req = github.apps.getInstallationRepositories({per_page: 100})
-    return github.paginate(req, res => {
-      const repos = res.data.repositories;
-      (options.filter ? repos.filter(repo => options.filter(installation, repo)) : repos)
-        .forEach(async repository => callback(repository, github));
-    })
+    app.log.trace({ installation }, 'Fetching repositories for installation')
+    const github = await app.auth(installation.id)
+
+    const repositories = (await github.paginate(
+      github.apps.listRepos.endpoint.merge({ per_page: 100 })
+    ))
+      .map(page => page.repositories)
+      .flat()
```

Comment:
I think using `.flat` will lock this to Node v11+? Might not matter but it seemed worth pointing out.

I randomly came across this while messing with a local installation of `probot/stale`, where I was using Node 10.

Feel free to dismiss if this isn't relevant.

Reply:
Oh good point! I definitely don't think we should lock it to a non-LTS release of Node. I originally was just going to do this in a `.reduce` method, but then realized I was basically recreating `flat()`, but I forgot that was new in Node 11.0